### PR TITLE
improve `sm_url` provider attribute documentation

### PR DESCRIPTION
**Pull request by mdb**
This seeks to update PR #571 with some potential language improvements (it also seeks to fix PR #571 's CI build).

### CODE REVIEW

**Comment by mdb** [comment]
@mem @inkel - Could this get PR #571 to a merge-able state? Alternatively, I could open this PR against `master`, though I didn't want to entirely bulldoze over @mem 's original contribution :)

File: docs/index.md [hunk 1]
```diff
@@ -221,7 +215,7 @@ resource "grafana_oncall_escalation" "example_notify_step" {
...
-- `sm_url` (String) Synthetic monitoring backend address. May alternatively be set via the `GRAFANA_SM_URL` environment variable. You can find the correct value for each service region [here](https://grafana.com/docs/grafana-cloud/synthetic-monitoring/private-probes/#probe-api-server-url), under the header "backend address".
+- `sm_url` (String) Synthetic monitoring backend address. May alternatively be set via the `GRAFANA_SM_URL` environment variable. The correct value for each service region is cited in the [Synthetic Monitoring documentation](https://grafana.com/docs/grafana-cloud/synthetic-monitoring/private-probes/#probe-api-server-url). Note the `sm_url` value is optional, but it must correspond with the value specified as the `region_slug` in the `grafana_cloud_stack` resource. Also note that when a Terraform configuration contains multiple provider instances managing SM resources associated with the same Grafana stack, specifying an explicit `sm_url` set to the same value for each provider ensures all providers interact with the same SM API.
```
Comment (mdb) [inline docs/index.md:218]: Note that this file's creation is automated via `go generate` and its contents are ultimately powered by the `Description` fields in the `grafana/provider.go` `*schema.Provider`.

**Review by inkel — APPROVED** [review]
Thanks @mdb for the contribution! This looks good to me, but I'd like to ask @mem for final approval.

**Comment by mdb** [comment]
Hi @mem - Does this PR address your concerns?

**Review by julienduchesne — APPROVED** [review]

**Pull request merged**